presto-kinesis/src/main/java/io/prestosql/plugin/kinesis/KinesisConnectorFactory.java

```diff
@@ -0,0 +1,168 @@
+            Map<String, String> optionalConfig,
+            Optional<Class<? extends KinesisClientProvider>> altProviderClass)
+    {
+        this.classLoader = classLoader;
+        this.tableDescriptionSupplier = requireNonNull(tableDescriptionSupplier, "tableDescriptionSupplier is null");
+        this.optionalConfig = requireNonNull(optionalConfig, "optionalConfig is null");
+        this.altProviderClass = requireNonNull(altProviderClass, "altProviderClass is null");
+
+        this.handleResolver = new KinesisHandleResolver(connectorName);
+
+        // Explanation: AWS uses a newer version of jackson (2.6.6) than airlift (2.4.4).  In order to upgrade
+        // to the latest version of the AWS API, we need to turn this feature off.  This can be set
+        // in jvm.properties but trying to make this more foolproof.
+        System.setProperty("com.amazonaws.sdk.disableCbor", "true");
+    }
+
+    @Override
+    public String getName()
+    {
+        return connectorName;
+    }
+
+    @Override
+    public ConnectorHandleResolver getHandleResolver()
+    {
+        return new KinesisHandleResolver(connectorName);
+    }
+
+    @Override
+    public Connector create(String connectorId, Map<String, String> config, ConnectorContext context)
+    {
+        log.info("In connector factory create method.  Connector id: " + connectorId);
+        requireNonNull(connectorId, "connectorId is null");
+        requireNonNull(config, "config is null");
+
+        try {
+            Bootstrap app = new Bootstrap(
+                    new JsonModule(),
+                    new KinesisConnectorModule(),
+                    binder -> {
+                        binder.bindConstant().annotatedWith(Names.named("connectorId")).to(connectorId);
+                        binder.bind(ConnectorId.class).toInstance(new ConnectorId(connectorId));
+                        binder.bind(TypeManager.class).toInstance(context.getTypeManager());
+                        binder.bind(NodeManager.class).toInstance(context.getNodeManager());
+                        // Note: moved creation from KinesisConnectorModule because connector manager accesses it earlier!
+                        binder.bind(KinesisHandleResolver.class).toInstance(new KinesisHandleResolver(connectorName));
+
+                        // Moved creation here from KinesisConnectorModule to make it easier to parameterize
+                        if (altProviderClass.isPresent()) {
+                            binder.bind(KinesisClientProvider.class).to(altProviderClass.get()).in(Scopes.SINGLETON);
+                        }
+                        else {
+                            binder.bind(KinesisClientProvider.class).to(KinesisClientManager.class).in(Scopes.SINGLETON);
+                        }
+
+                        if (tableDescriptionSupplier.isPresent()) {
```

Comment:
This seems brittle and makes the code harder to understand. Also, since it's used only in tests it results in some code paths not being exercised during testing. It'd be cleaner to go through the standard mechanism of providing a URL/file that contains the description of the schemas/tables and let the plugin parse and handle them as it would during regular usage.

Reply:
Changed code so that table descriptors are not accessed outside and rewrote tests using temp json files